### PR TITLE
Add configuration for GCP webhook authorization.

### PR DESCRIPTION
**Pull request by cjcullen**
Plumb through configuration of webhook authorization on GCE.

### CODE REVIEW

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -798,6 +798,29 @@ EOF
     rm -f /etc/gce.conf
   fi
 
+  if [[ -n ${GCP_AUTHZ_URL:-} ]]; then
```
Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: nit: quotes: `"${GCP_AUTHZ_URL:-}"`

**Comment by zmerlynn** [comment]
LGTM modulo nit.

File: cluster/gce/configure-vm.sh (hunk 1 above)
Comment (cjcullen) [inline cluster/gce/configure-vm.sh]: Done.

**Comment by fgrzadkowski** [comment]
GKE jenkins job are broken since roughly the time of this PR being merged. I'm gonna revert it to verify if this helps.

**Comment by andyzheng0831** [comment]
Just had an offline discussion with @cjcullen. I will make GCI side change, cherry pick CJ's this PR and #25380 into my change. Then we can have the support on both distro merged at the same time.

**Pull request merged**